### PR TITLE
Update patricia-merkle-trie/index.md

### PR DESCRIPTION
**Pull request by gichiba**
This edit improves the introduction to the Merkle-Patricia trie article. 

## Description

The EF website may soon link to this page, so an introductory paragraph pitched more to a 'general audience' was needed to give curious individuals a softer landing, so to speak. 

Also, the previous version's introduction made a claim about the 'efficiency' of inserts, lookups, and deletes,  as well as a comparison to a data structure not used in Ethereum, which doesn't seem relevant for an introduction. The claim is a bit misleading, as the more compact trie structure chosen for protocol efficiency has actually created a bigger headache for execution client implementations -- better to just focus on the more straightforward description of the trie itself and leave discussions of efficiency for another article.

### CODE REVIEW

File: src/content/developers/docs/data-structures-and-encoding/patricia-merkle-trie/index.md [hunk 1]
```diff
@@ -5,13 +5,15 @@ lang: en
...
 
 ## Prerequisites {#prerequisites}
 
-To better understand this page, it would be helpful to have basic knowledge of [hashes](https://en.wikipedia.org/wiki/Hash_function), [Merkle trees](https://en.wikipedia.org/wiki/Merkle_tree), [tries](https://en.wikipedia.org/wiki/Trie) and [serialization](https://en.wikipedia.org/wiki/Serialization).
+To better understand Ethereum's modified Merkle-Patricia trie, it would be helpful to have basic knowledge of [hashes](https://en.wikipedia.org/wiki/Hash_function), [Merkle trees](https://en.wikipedia.org/wiki/Merkle_tree), [tries](https://en.wikipedia.org/wiki/Trie) and [serialization](https://en.wikipedia.org/wiki/Serialization). This article begins with a description of a basic [radix tree](https://en.wikipedia.org/wiki/Radix_tree), then gradually introduces the modifications necessary for Ethereum's more optimized data structure.
```
Comment (wackerow) [inline src/content/developers/docs/data-structures-and-encoding/patricia-merkle-trie/index.md:16]: Not necessarily against this but it breaks the pattern used in several other dev doc pages of saying "to better understand this page"

> This article begins with a description of a basic [radix tree](https://en.wikipedia.org/wiki/Radix_tree), then gradually introduces the modifications necessary for Ethereum's more optimized data structure. 

Wouldn't put this in pre-req's... maybe pull up to the end of the intro.

Comment (wackerow) [inline src/content/developers/docs/data-structures-and-encoding/patricia-merkle-trie/index.md:16]: Not necessarily against this but it breaks the pattern used in several other dev doc pages of saying "to better understand this page"

> This article begins with a description of a basic [radix tree](https://en.wikipedia.org/wiki/Radix_tree), then gradually introduces the modifications necessary for Ethereum's more optimized data structure. 

Wouldn't put this in pre-req's... maybe pull up to the end of the intro.

**Review by wackerow — COMMENTED** [review]
Thanks @gichiba! Overall like the updates

@minimalsm would love your take on new sentence structure here

`re**trie**val` is a nice touch, but wondering if this will cause difficulties with translations? cc: @lukassim 

Any chance we could have the $O(log(n))$ notation added back in somewhere?

File: src/content/developers/docs/data-structures-and-encoding/patricia-merkle-trie/index.md (hunk 1 above)
Comment (gichiba) [reply src/content/developers/docs/data-structures-and-encoding/patricia-merkle-trie/index.md]: Okay, can rephrase to use the consistent language of other pages

**Comment by gichiba** [comment]
Hey @wackerow sorry I let this sit for so long; was on a long holiday leave but back in the chair! 

I have re-based this branch and updated the proposed changes based on your comments above!

**Review by wackerow — APPROVED** [review]

**Pull request merged**